admin/services/data-access/check-window.data.service.js

```diff
@@ -0,0 +1,84 @@
+'use strict'
+
+const moment = require('moment')
+
+const CheckWindow = require('../../models/check-window')
+
+const checkWindowDataService = {
+
+  /**
+   * Fetch check window document by id.
+   * @param id
+   * @returns {Promise.<*>}
+   */
+  fetchCheckWindow: async (id) => {
+    let checkWindow
+    checkWindow = await CheckWindow.findOne({'_id': id, 'deleted': false}).exec()
+    return checkWindow
+  },
+  /**
+   * Set check window as deleted.
+   * @param id
+   * @returns {Promise.<void>}
+   */
+  setDeletedCheckWindow: async (id) => {
+    let checkWindow
+    checkWindow = await CheckWindow.updateOne({'_id': id}, {$set: {'deleted': true}}).exec()
+    return checkWindow
+  },
+  /**
+   * Fetch check windows by status, sort by, sort direction and date (current or past).
+   * @param deleted
+   * @param sortBy
+   * @param sortDirection
+   * @param current
+   * @returns {Promise.<void>}
+   */
+  fetchCheckWindows: async (sortBy, sortDirection, deleted, current) => {
+    return new Promise(async (resolve, reject) => {
```

Comment:
I don't think there is a need to return a Promise when in an async function.  Returns or thrown errors will be converted to resolve and reject calls automatically.

Reply:
OK. Then perhaps we should modify this pattern everywhere else it's been used too.